incubator/hnc/pkg/reconcilers/hierarchical_namespace.go

```diff
@@ -59,51 +59,97 @@ func (r *HierarchicalNamespaceReconciler) Reconcile(req ctrl.Request) (ctrl.Resu
 	nm := req.Name
 	pnm := req.Namespace
 
-	// Get instance from apiserver. If the instance doesn't exist, we don't want to reconcile
-	// it since it may trigger the HC reconciler to recreate the namespace that was just deleted.
-	// TODO expand on this to check the owner's hc.spec.allowCascadingDelete. If it's set to
-	//  true, we still want to reconcile the hns instance. See issue:
-	//  https://github.com/kubernetes-sigs/multi-tenancy/issues/501
+	// Get instance from apiserver. If the instance doesn't exist, we don't want
+	// to reconcile it to create the HNS instance or the namespace.
```

Comment:
I'm not sure I understand this comment. If it doesn't exist, it means it's been deleted - how should we handle deletions?

Reply:
Updated with 'If the instance doesn't exist, do nothing and early exist because HCR watches HNS instances and (if applicable) has already updated the owner HC when this HNS was purged.'